velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +325,86 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  int precision = getDecimalPrecisionScale(*fromType).first;
+  int scale = getDecimalPrecisionScale(*fromType).second;
+  // A varchar's size is estimated with unscaled value digits, dot, leading
+  // zero, and possible minus sign.
+  int varcharSize = precision + !!(scale > 0) + !!(precision == scale) + 1;
```

Comment:
+ !!(precision == scale && scale > 0)

Reply:
`Scale > 0` is for the dot and `precision == scale` is for the leading zero. Extra `1` is for possible minus sign because we do not need to check nagativitity for every input value. Therefore, we need to get 8 instead of 7 for a number like 0.12345 (decimal(5, 5)).

Reply:
Only if has dot, we may need to add leading zeros. 0.x is valid, .x does not exists.

Reply:
Looks that only happens when scale and precision are both zero. We may not need to double-check `scale > 0` just for this corner case, because a larger estimated size is acceptable.

Reply:
What's your opinion? @mbasmanova  Do we need to add this optimization for case `precision =0 && scale ==0`?

Reply:
Yes, I agree.